src/serializationHelpers.ts

```diff
@@ -173,3 +327,582 @@ export function serializeYearMonth(yearMonth: string): Buffer {
+                    const fieldInfo = namedFields.contents[i];
+                    const userJsonProperty = fieldInfo[0];
+                    const userValue = userData[userJsonProperty];
+                    bufferStruct.push(
+                        serializeParameters(fieldInfo[1], userValue)
+                    );
+                }
+                return Buffer.concat(bufferStruct);
+            } else {
+                throw new Error(
+                    `Expected ${namedFields.contents.length} named fields`
+                );
+            }
+        case FieldsTag.Unnamed:
+            const unNamedFields = fields as UnNamedFields;
+            if (userData.length === unNamedFields.contents.length) {
+                for (let i = 0; i < unNamedFields.contents.length; i++) {
+                    const fieldInfo = unNamedFields.contents[i];
+                    const userValue = userData[i];
+                    bufferStruct.push(
+                        serializeParameters(fieldInfo, userValue)
+                    );
+                }
+                return Buffer.concat(bufferStruct);
+            } else {
+                throw new Error(
+                    `Expected ${unNamedFields.contents.length} unnamed fields`
+                );
+            }
+        case FieldsTag.None:
+        default:
+            return Buffer.from(bufferStruct);
+    }
+}
+
+/**
+ *
+ * @param length length of the values provided by the user
+ * @param sizeLength sizeLength represented as an unsigned integer
+ * @returns
+ */
+
+export function serializeLength(
+    length: number,
+    sizeLength: SizeLength
+): Buffer {
+    switch (sizeLength) {
+        case SizeLength.U8:
+            return encodeWord8(length);
+        case SizeLength.U16:
+            return encodeWord16(length, true);
+        case SizeLength.U32:
+            return encodeWord32(length, true);
+        case SizeLength.U64:
+            return encodeWord64(BigInt(length), true);
+        default:
+            return Buffer.from([]);
+    }
+}
+
+/**
+ *
+ * @param value Duration string
+ * @returns milliseconds value for the given duration
+ */
+export function getMilliSeconds(value: string): number {
+    let seconds = 0;
+    const days = getDays(value);
+    const hours = getHours(value);
+    const minutes = getMinutes(value);
+    const sec = getSeconds(value);
+    const millisec = getMSeconds(value);
+    let ms = 0;
+    if (days) {
+        seconds += days * 86400;
+    }
+    if (hours) {
+        seconds += hours * 3600;
+    }
+    if (minutes) {
+        seconds += minutes * 60;
+    }
+    if (sec) {
+        seconds += sec;
+    }
+    if (millisec) {
+        ms = millisec;
+    }
+    seconds *= 1000;
+    seconds = seconds + ms;
+    return Math.round(seconds);
+}
+
+/**
+ *
+ * @param value  Duration string
+ * @returns sum of hours in the given string
+ */
+export function getHours(value: string): number {
```

Comment:
I feel like these `get{Duration part}` functions could merged into one by supplying the symbol as a parameter :thinking: 
Also I don't think they should be exported, as their purpose is to be subroutines for getMilliSeconds?